### ISSUE TITLE
Runtime panic

### ISSUE THREAD

**Issue by chrisfentiman** [eawsy/aws-lambda-go-shim#29]
Hey,

I am going crazy trying to figure this panic out. It doesn't happen all the time but if I run a small test with 1000 requests per second against the lambda function at-least 100 fail with this result.

```
unexpected fault address 0x0
fatal error: fault
[signal SIGSEGV: segmentation violation code=0x80 addr=0x0 pc=0x7faf0b42df68]

goroutine 17 [running, locked to thread]:
runtime.throw(0x7faf0b4b02d0, 0x5)
/usr/local/go/src/runtime/panic.go:596 +0x97 fp=0xc4201a78f0 sp=0xc4201a78d0
runtime.sigpanic()
/usr/local/go/src/runtime/signal_unix.go:297 +0x290 fp=0xc4201a7940 sp=0xc4201a78f0
runtime.call64(0xc4201dbf50, 0x7faf038aa9c0, 0xc4201dbfb0, 0x1000000030)
/usr/local/go/src/runtime/asm_amd64.s:515 +0x48 fp=0xc4201a7990 sp=0xc4201a7940
reflect.Value.call(0x7faf03c59d20, 0x7faf038aa9c0, 0x13, 0x7faf0b4b0078, 0x4, 0xc4201a7e10, 0x2, 0x2, 0x7faf0b6e29c0, 0x7faf0b70a368, ...)
/usr/local/go/src/reflect/value.go:434 +0x921 fp=0xc4201a7cd8 sp=0xc4201a7990
reflect.Value.Call(0x7faf03c59d20, 0x7faf038aa9c0, 0x13, 0xc4201a7e10, 0x2, 0x2, 0xc4200726f8, 0x193, 0x7faf0b406b48)
/usr/local/go/src/reflect/value.go:302 +0xa6 fp=0xc4201a7d40 sp=0xc4201a7cd8
main.handle(0x23e29f4, 0x2332eb4, 0x23e23a4, 0xc4201a7eb0, 0xc4201a7eb8, 0x0)
/build/src/runtime.go:140 +0x4e1 fp=0xc4201a7e50 sp=0xc4201a7d40
main._cgoexpwrap_dd780bcacc70_handle(0x23e29f4, 0x2332eb4, 0x23e23a4, 0x0, 0x0, 0x0)
_/build/src/_obj/_cgo_gotypes.go:127 +0x9a fp=0xc4201a7e90 sp=0xc4201a7e50
runtime.call64(0x0, 0x7ffc8d2d1728, 0x7ffc8d2d17c0, 0x30)
/usr/local/go/src/runtime/asm_amd64.s:515 +0x4a fp=0xc4201a7ee0 sp=0xc4201a7e90
runtime.cgocallbackg1(0x0)
/usr/local/go/src/runtime/cgocall.go:301 +0x1a1 fp=0xc4201a7f58 sp=0xc4201a7ee0
runtime.cgocallbackg(0x0)
/usr/local/go/src/runtime/cgocall.go:184 +0x86 fp=0xc4201a7fc0 sp=0xc4201a7f58
runtime.cgocallback_gofunc(0x0, 0x0, 0x0, 0x0)
/usr/local/go/src/runtime/asm_amd64.s:767 +0x71 fp=0xc4201a7fe0 sp=0xc4201a7fc0
runtime.goexit()
/usr/local/go/src/runtime/asm_amd64.s:2197 +0x1 fp=0xc4201a7fe8 sp=0xc4201a7fe0

goroutine 18 [syscall, locked to thread]:
runtime.goexit()
/usr/local/go/src/runtime/asm_amd64.s:2197 +0x1
```

This panic seems to be happening before executing any of my code. If anything I can trace it to the handle function inside the src/runtime.go

**Comment by lsuss** [comment]
Hi @chrisfentiman ,

First of all, thank you for your issue, we understand pretty well how it can turn you crazy since we experienced an error like that in the past (we fixed it and/or created an issue on golang).

To be able to help you we need the **EXACT** code you're running, could you post it here ? Or maybe in private to @fsenart or me on https://gitter.im/eawsy/bavardage.

**Comment by lsuss** [comment]
Oh and please check this issue:
golang/go#19269

As mentionned here:
https://github.com/eawsy/aws-lambda-go-shim#known-limitations

**Comment by chrisfentiman** [comment]
Ahh, you're so right. 

Sorry for over looking the previous post.

I was generating an extra file which only created exported functions (Not thinking that this actually counted). I will change the tool on my end and it should work fine.

Thanks again for responding so fast.

**Comment by lsuss** [comment]
You are welcome :).

For your information, this bug should be fixed in 1.9.
As soon as 1.9 is released, we will up the go version and make some tests to be sure this limitation isn't a problem anymore.

Keep connected 👍
